app/src/components/ToggleView/ToggleView.tsx

```diff
@@ -0,0 +1,48 @@
+import { ViewState } from '../../types/view';
+import './Toggle.scss';
+
+const iconPaths: Record<ViewState, string> = {
+    grid: '/src/assets/Icons-Grid.svg',
+    list: '/src/assets/Icons-List.svg'
+};
+
+function ToggleButton({ currentView, onToggle }: ToggleViewProps) {
+    return (
+        <button
+            className="toggle-button"
+            onClick={() => onToggle(currentView)}>
+            <img className="toggle-icon"
+                src={iconPaths[currentView]}
+                alt={`${currentView} view`}
+            />
+        </button>
+    );
+}
+
+function ToggleSelector({ currentView }: { currentView: ViewState }) {
+    return (
+        <img
+            className="toggle-selector"
+            data-view={currentView}
+            src="/src/assets/Ellipse-Selector.svg"
+            alt="Selector"
+        />
+    );
+}
```

Comment:
In my opinion, there's limited value in encapsulating an `image` element within a component. Consider incorporating it directly in the sole location where it's used.